src/dex/clear/clear.ts

```diff
@@ -0,0 +1,623 @@
+        const vaultAddress = poolIdentifier.split('_')[1];
+
+        // Add calls for each amount
+        for (let i = 0; i < amounts.length; i++) {
+          const callData = this.clearSwapIface.encodeFunctionData(
+            'previewSwap',
+            [vaultAddress, srcToken.address, destToken.address, amounts[i]],
+          );
+
+          calls.push({
+            target: this.config.swapAddress,
+            callData,
+            decodeFunction: (result: any) => {
+              const decoded = this.clearSwapIface.decodeFunctionResult(
+                'previewSwap',
+                result,
+              );
+              return {
+                amountOut: BigInt(decoded.amountOut.toString()),
+                ious: BigInt(decoded.ious.toString()),
+              };
+            },
+          });
+
+          callInfos.push({
+            vaultAddress,
+            poolIdentifier,
+            isUnit: false,
+            amountIndex: i,
+          });
+        }
+
+        // Add call for unit price
+        const unitCallData = this.clearSwapIface.encodeFunctionData(
+          'previewSwap',
+          [vaultAddress, srcToken.address, destToken.address, unitAmount],
+        );
+
+        calls.push({
+          target: this.config.swapAddress,
+          callData: unitCallData,
+          decodeFunction: (result: any) => {
+            const decoded = this.clearSwapIface.decodeFunctionResult(
+              'previewSwap',
+              result,
+            );
+            return {
+              amountOut: BigInt(decoded.amountOut.toString()),
+              ious: BigInt(decoded.ious.toString()),
+            };
+          },
+        });
+
+        callInfos.push({
+          vaultAddress,
+          poolIdentifier,
+          isUnit: true,
+          amountIndex: -1,
+        });
+      }
+
+      // Execute all calls in a single multicall batch
+      const results = await this.dexHelper.multiWrapper.tryAggregate<{
+        amountOut: bigint;
+        ious: bigint;
+      }>(
+        false, // mandatory = false, allow partial failures
+        calls,
+        blockNumber,
+        this.dexHelper.multiWrapper.defaultBatchSize,
+        false, // reportFails = false, we handle failures ourselves
+      );
+
+      // Group results by vault
+      const vaultResults: Map<
+        string,
+        {
+          poolIdentifier: string;
+          prices: bigint[];
+          unit: bigint;
+        }
+      > = new Map();
+
+      for (let i = 0; i < results.length; i++) {
+        const result = results[i];
+        const info = callInfos[i];
+
+        if (!vaultResults.has(info.vaultAddress)) {
+          vaultResults.set(info.vaultAddress, {
+            poolIdentifier: info.poolIdentifier,
+            prices: new Array(amounts.length).fill(0n),
+            unit: 0n,
+          });
+        }
+
+        const vaultData = vaultResults.get(info.vaultAddress)!;
+
+        if (result.success) {
```

Comment:
A lot of cyclomatic complexity here

Maybe extract those if elses in a func and use short returns in order to make this more digest ?